types/react-numeric-input/index.d.ts

```diff
@@ -9,38 +10,52 @@ import * as React from 'react';
 export = NumericInput;
 
 declare namespace NumericInput {
-    interface NumericInputProps extends React.Props<NumericInput> {
+    type BoundsFunctionProp = number | ((component: NumericInput) => number | undefined);
```

Comment:
I'm curious why some of the functions return `| undefined` (or null)? And for example, why `max` can return `number | undefined` but `precision` can return `number | null | undefined` (ie, they don't all include `null`)?

Reply:
A return value of `undefined` will use whatever is a built-in default for the library (min's default is `Number.MIN_SAFE_INTEGER`, max's default is `Number.MAX_SAFE_INTEGER`, `step`'s is 1, and `precision`'s default is `null`). 

That `null` value for `precision()` is used in two different ways.

* [if `precision()` returns null, 10 is used](https://github.com/vlad-ignatov/react-numeric-input/blob/79874ccbe2293873300953f32d91f2d70d32a871/src/NumericInput.jsx#L611) when parsing in strict mode
* [if `precision()` returns null, `toFixed` is not called at all](https://github.com/vlad-ignatov/react-numeric-input/blob/79874ccbe2293873300953f32d91f2d70d32a871/src/NumericInput.jsx#L645-L647) when formatting numbers for display